container.go

```diff
@@ -281,6 +281,7 @@ func (c *container) NewTask(ctx context.Context, ioCreate cio.Creator, opts ...N
 			})
 		}
 	}
+	request.RuntimePath = info.RuntimePath
```

Comment:
Can we have a test?

Reply:
I added a test, in linux it is green, but in windows it cannot find the shim binary.
https://github.com/containerd/containerd/actions/runs/4414423604/jobs/7736233010#step:16:37

```
    container_test.go:269: failed to start shim: start failed: : exec: "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\shim1680046787": file does not exist: unknown
```

Reply:
@AkihiroSuda It's green on windows and linux, PTAL